editor/debugger/script_editor_debugger.cpp

```diff
@@ -743,18 +743,19 @@ void ScriptEditorDebugger::_performance_draw() {
 
 	info_message->hide();
 
-	Ref<StyleBox> graph_sb = get_theme_stylebox("normal", "TextEdit");
-	Ref<Font> graph_font = get_theme_font("font", "TextEdit");
+	const Ref<StyleBox> graph_sb = get_theme_stylebox("normal", "TextEdit");
+	const Ref<Font> graph_font = get_theme_font("font", "TextEdit");
 
-	int cols = Math::ceil(Math::sqrt((float)which.size()));
+	const int cols = Math::ceil(Math::sqrt((float)which.size()));
 	int rows = Math::ceil((float)which.size() / cols);
 	if (which.size() == 1) {
 		rows = 1;
 	}
 
-	int margin = 3;
-	int point_sep = 5;
-	Size2i s = Size2i(perf_draw->get_size()) / Size2i(cols, rows);
+	constexpr int margin = 3;
+	constexpr int point_sep = 5;
```

Comment:
Why are these `constexpr` and not just `const`?

Also, we should likely start a broader discussion on whether this kind of micro optimizations actually make a difference at all. If yes, we should use them consistently in new code. If not, we should probably not refactor existing code to use it.

Reply:
I changed them to use `const`. And yes, we should probably look at using `constexpr` instead of symbolic `#define`s in the future.

Reply:
I mean, does using `const` for local variables really make a performance difference? Or are compilers clever enough to optimize all this code already and us doing it manually is overkill? I think this should be benchmarked before continuing to make all local variables `const`.

Reply:
I think it's more about code readability than performance. It's likely that `const` is rarely useful with local variables in compiler optimization. However, with `const`, it's guaranteed the variable isn't being reassigned below. This can make the code significantly easier to read and less error-prone. I've seen lots of old code in Godot where variables are being reassigned for no good reason, and that makes it harder to reason about the code.

The only case where I consider `const` to be overkill is in function parameters for primitive types ([like the ISO C++ Core Guidelines say](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-const)).